runners/spark/pom.xml

```diff
@@ -120,6 +120,24 @@
       <artifactId>hamcrest-all</artifactId>
       <scope>test</scope>
     </dependency>
+    <!-- Depend on test jar to scan for RunnableOnService tests -->
+    <dependency>
+      <groupId>org.apache.beam</groupId>
+      <artifactId>java-sdk-all</artifactId>
+      <classifier>tests</classifier>
+      <scope>test</scope>
+      <exclusions>
+        <exclusion>
+          <groupId>org.slf4j</groupId>
+          <artifactId>slf4j-jdk14</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>org.mockito</groupId>
```

Comment:
Ah, yes, I think I see. To get the behavior automatically, we would probably need a `java-sdk-tests` artifact with a `compile`-scoped dependency for everything that is currently a `test`-scoped dependency. I propose solving this in the future instead of now.